src/components/course/data/utils.jsx

```diff
@@ -679,7 +679,7 @@ export const getSubscriptionDisabledEnrollmentReasonType = ({
   const subscriptionsApplicableToCourse = customerAgreementConfig?.subscriptions?.filter(
     subscription => catalogsWithCourse.includes(subscription?.enterpriseCatalogUuid),
   );
-  const hasApplicableSubscriptions = subscriptionsApplicableToCourse.length > 0;
+  const hasApplicableSubscriptions = subscriptionsApplicableToCourse?.length > 0;
```

Comment:
nit: given this is a boolean variable (implied by the name `has*`), consider casting the value to a boolean so that if `subscriptionsApplicableToCourse` is undefined, `hasApplicableSubscriptions` is still a boolean value (false) as opposed to `undefined` (though still falsey).